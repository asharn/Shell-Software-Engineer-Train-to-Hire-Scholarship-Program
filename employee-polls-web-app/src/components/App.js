import NavBar from './NavBar';
import { useEffect} from "react";
import { connect } from "react-redux";
import { handleInitialData } from "../actions/shared";
import LeaderBoard from "./LeaderBoard";
import SignUp from "./SignUp";
import SignIn from "./SignIn";
import Home from "./Home";
import NewQuestion from "./NewQuestion";
import { Routes, Route } from "react-router-dom";
import PageNotFound from './PageNotFound';
import NotAuthorized from './NotAuthorized';
import Copyright from './Copyright';

const App = (props) => {
   useEffect(() => {
     props.dispatch(handleInitialData());
   }, []);
    console.log(props.userLoggedIn);
    console.log(props);
  return (
      <div className="container">
        <NavBar />
        {props.userLoggedIn !== true ?  (
          <Routes>
            <Route path='/' exact element={<Home />} />
            <Route path='/leaderboard' exact element={<LeaderBoard />} />
            <Route path='/add' exact element={<NewQuestion/>} />
            <Route path='/sign-up' exact element={<SignUp/>} />
            <Route path='/sign-in' element={<SignIn/>} />
            <Route path="*" element={<PageNotFound />} />
          </Routes>
        ):
        (
          <Routes>
            <Route path='/sign-in' element={<SignIn/>} />
            <Route path='/sign-up' exact element={<SignUp/>} />
            <Route path='/leaderboard' exact element={<NotAuthorized />} />
            <Route path='/add' exact element={<NotAuthorized/>} />
            <Route path='/' exact element={<NotAuthorized />} />
            <Route path="*" element={<PageNotFound />} />
          </Routes>
        )}
        <Copyright sx={{ mt: 4, mb: 10, backgroundColor: 'primary' }} />
      </div>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  userLoggedIn: authedUser !== null,
});
export default connect(mapStateToProps)(App);