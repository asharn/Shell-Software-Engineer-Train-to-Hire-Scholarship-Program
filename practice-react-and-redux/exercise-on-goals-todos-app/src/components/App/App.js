import { useEffect } from "react";
import ConnectedTodos from "../Todos/Todos";
import ConnectedGoals from "../Goals/Goals";
import { connect } from "react-redux";
import { handleInitialData } from "../../actions/shared";

const App = (props) => {
  useEffect(() => {
    props.dispatch(handleInitialData());
  }, []);

  if (props.loading === true) {
    return <h3>Loading</h3>;
  }

  return (
    <div>
      <ConnectedTodos />
      <ConnectedGoals />
    </div>
  );
};

export default connect((state) => ({
  loading: state.loading,
}))(App);