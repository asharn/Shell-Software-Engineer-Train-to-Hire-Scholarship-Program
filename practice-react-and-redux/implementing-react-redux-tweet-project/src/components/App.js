import { useEffect, Fragment } from "react";
import { connect } from "react-redux";
import { handleInitialData } from "../actions/shared";
import Dashboard from "./Dashboard";
import LoadingBar from "react-redux-loading-bar";
import NewTweet from "./NewTweet";
import TweetPage from "./TweetPage";
import Nav from "./Nav";
import { Routes, Route } from "react-router-dom";

const App = (props) => {
  useEffect(() => {
    props.dispatch(handleInitialData());
  }, []);

  return (
    <Fragment>
      <LoadingBar />
      <div className="container">
        <Nav />
        {props.loading === true ? null : (
          <Routes>
            <Route path="/" exact element={<Dashboard />} />
            <Route path="/tweet/:id" element={<TweetPage />} />
            <Route path="/new" element={<NewTweet />} />
          </Routes>
        )}
      </div>
    </Fragment>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  loading: authedUser === null,
});
export default connect(mapStateToProps)(App);