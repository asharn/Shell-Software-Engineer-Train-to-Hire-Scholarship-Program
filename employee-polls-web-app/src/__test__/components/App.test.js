import App from '../../components/App';
import { createStore } from 'redux';
import rootReducer from '../../reducers';
import { connect } from "react-redux";

describe('Snapshot testing for App component for logged in user as well as logout use.', () => {
  test('renders App component for LoggedIn User.', async () => {
    const store = await createStore(rootReducer);
    const mapStateToProps = (state) => {
      let {authedUser} = state
      return {authedUser};
    }
    const render = connect(store, App); // reusable with
    const component = render(mapStateToProps); 
    expect(component).toMatchSnapshot()
  });
})