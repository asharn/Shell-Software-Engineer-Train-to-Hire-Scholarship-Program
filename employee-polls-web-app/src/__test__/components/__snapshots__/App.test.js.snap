// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for App component for logged in user as well as logout use. renders App component for LoggedIn User. 1`] = `
Object {
  "$$typeof": Symbol(react.memo),
  "WrappedComponent": [Function],
  "compare": null,
  "type": [Function],
}
`;
